From aa6b9da8eb65e42001f22ba471f5864069eb4515 Mon Sep 17 00:00:00 2001
From: "spike.liao" <spike.liao@amlogic.com>
Date: Mon, 12 Jan 2015 10:52:06 +0800
Subject: [PATCH 5320/5965] pd 98750: if hdmiin display on video, cannot stop
 tsync when unreg amvideo

Squashed commit of the following:

commit 7fbf2cd7d2f1a067086c4a53d30604d13fae81b5
Author: lianghu.su <lianghu.su@amlogic.com>
Date:   Tue Oct 28 17:16:48 2014 +0800

    pd 98750: HDMIIN do not use freerun mode. patch from Spike.

    Change-Id: I366bd60162020ac56ba332444fa029ab7872f6dc

commit 43468a8fd462d1c38d6005a5f5cab0c37015e92c
Author: lianghu.su <lianghu.su@amlogic.com>
Date:   Tue Oct 28 16:27:28 2014 +0800

    pd 98750: if hdmiin display on video, cannot stop tsync when unreg amvideo. patch from spike.

    Change-Id: I24719043568027e586387908bbc0841d47c6a88d

Change-Id: I4780065d019ffd6be88b6d5221041a474fa24926
---
 drivers/amlogic/amports/video.c | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/drivers/amlogic/amports/video.c b/drivers/amlogic/amports/video.c
index 54aa054d604a..aa0bdd9af388 100755
--- a/drivers/amlogic/amports/video.c
+++ b/drivers/amlogic/amports/video.c
@@ -732,6 +732,7 @@ static int vpu_mem_power_off_count;
 static u32 vpts_ref = 0;
 static u32 video_frame_repeat_count = 0;
 static u32 smooth_sync_enable = 0;
+static u32 hdmi_in_onvideo = 0;
 #ifdef CONFIG_AM_VIDEO2
 static int video_play_clone_rate = 60;
 static int android_clone_rate = 30;
@@ -2492,7 +2493,7 @@ static inline bool vpts_expire(vframe_t *cur_vf, vframe_t *next_vf)
         return true;
     }
 
-    if (FREERUN_NODUR == freerun_mode) {
+    if (FREERUN_NODUR == freerun_mode || hdmi_in_onvideo) {
         return true;
     }
 
@@ -2994,7 +2995,8 @@ static irqreturn_t vsync_isr(int irq, void *dev_id)
         vf = video_vf_peek();
 
         if (vf) {
-            tsync_avevent_locked(VIDEO_START,
+            if (hdmi_in_onvideo == 0)
+                tsync_avevent_locked(VIDEO_START,
                           (vf->pts) ? vf->pts : timestamp_vpts_get());
 
 	     if(show_first_frame_nosync)
@@ -3799,14 +3801,18 @@ static void video_vf_unreg_provider(void)
         vf_keep_current();
         switch_mod_gate_by_name("ge2d", 0);
     }
-    tsync_avevent(VIDEO_STOP, 0);
+    if(hdmi_in_onvideo == 0){
+        tsync_avevent(VIDEO_STOP, 0);
+    }
 #else
     //if (!trickmode_fffb)
     if (cur_dispbuf)
     {
         vf_keep_current();
     }
-    tsync_avevent(VIDEO_STOP, 0);
+    if(hdmi_in_onvideo == 0){
+        tsync_avevent(VIDEO_STOP, 0);
+    }
  #endif
     atomic_set(&video_unreg_flag, 0);
     enable_video_discontinue_report = 1;
@@ -6375,6 +6381,9 @@ module_exit(video_exit);
 MODULE_PARM_DESC(smooth_sync_enable, "\n smooth_sync_enable\n");
 module_param(smooth_sync_enable, uint, 0664);
 
+MODULE_PARM_DESC(hdmi_in_onvideo, "\n hdmi_in_onvideo\n");
+module_param(hdmi_in_onvideo, uint, 0664);
+
 #ifdef CONFIG_AM_VIDEO2
 MODULE_PARM_DESC(video_play_clone_rate, "\n video_play_clone_rate\n");
 module_param(video_play_clone_rate, uint, 0664);
-- 
2.19.0

